src/runtime/prte_globals.c

```diff
@@ -551,6 +553,14 @@ static void prte_job_destruct(prte_job_t *job)
         PMIX_RELEASE(proc);
     }
 
+    if (prte_get_attribute(&job->attributes, PRTE_JOB_INFO_CACHE, (void **) &cache, PMIX_POINTER))
+    {
+        while (NULL != (kv = (prte_info_item_t *) pmix_list_remove_first(cache))) {
```

Comment:
We actually have a macro for doing this - just use `PMIX_LIST_RELEASE(cache)` and remove the `PMIX_RELEASE(cache)` line below.